ledger/simulation/simulator.go

```diff
@@ -197,12 +197,15 @@ func (s Simulator) evaluate(hdr bookkeeping.BlockHeader, stxns []transactions.Si
 	}
 
 	// Finally, process any pending end-of-block state changes.
-	vb, err := eval.GenerateBlock()
+	ub, err := eval.GenerateBlock(nil) // XXX not fetching proposer addresses
 	if err != nil {
 		return nil, err
 	}
 
-	return vb, nil
+	// XXX not setting seed & proposer details with FinishBlock/WithProposer
+	vb := ledgercore.MakeValidatedBlock(ub.UnfinishedBlock(), ub.UnfinishedDeltas())
```

Comment:
Does `ledgercore.ValidatedBlock` only exist for weird thigns like tests and simulation now? Because normally you'd have an `agreement.ValidatedBlock` that was actually an `agreement.proposableBlock`?

Reply:
Yes, I see now, it's the `ledgercore`'s ProposableBlock, in a way.